data/mods/gen9ssb/conditions.ts

```diff
@@ -1227,6 +1239,90 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			this.add('-end', pokemon, 'Attract', '[silent]');
 		},
 	},
+
+	gravity: {
+		inherit: true,
+		duration: 5,
+		durationCallback(source, effect) {
+			if (source?.hasAbility('persistent')) {
+				this.add('-activate', source, 'ability: Persistent', '[move] Gravity');
+				return 7;
+			}
+			return 5;
+		},
+		onFieldStart(target, source) {
+			if (source?.hasAbility('persistent')) {
+				this.add('-fieldstart', 'move: Gravity', '[persistent]');
+			} else {
+				this.add('-fieldstart', 'move: Gravity');
+			}
+			for (const pokemon of this.getAllActive()) {
+				let applies = false;
+				if (pokemon.removeVolatile('bounce') || pokemon.removeVolatile('fly')) {
+					applies = true;
+					this.queue.cancelMove(pokemon);
+					pokemon.removeVolatile('twoturnmove');
+				}
+				if (pokemon.volatiles['skydrop']) {
+					applies = true;
+					this.queue.cancelMove(pokemon);
+
+					if (pokemon.volatiles['skydrop'].source) {
+						this.add('-end', pokemon.volatiles['twoturnmove'].source, 'Sky Drop', '[interrupt]');
+					}
+					pokemon.removeVolatile('skydrop');
+					pokemon.removeVolatile('twoturnmove');
+				}
+				if (pokemon.volatiles['magnetrise']) {
+					applies = true;
+					delete pokemon.volatiles['magnetrise'];
+				}
+				if (pokemon.volatiles['telekinesis']) {
+					applies = true;
+					delete pokemon.volatiles['telekinesis'];
+				}
+				if (applies) this.add('-activate', pokemon, 'move: Gravity');
+			}
+		},
+		onModifyAccuracy(accuracy) {
+			if (typeof accuracy !== 'number') return;
+			return this.chainModify([6840, 4096]);
+		},
+		onDisableMove(pokemon) {
+			for (const moveSlot of pokemon.moveSlots) {
+				if (this.dex.moves.get(moveSlot.id).flags['gravity']) {
+					pokemon.disableMove(moveSlot.id);
+				}
+			}
+		},
+		// groundedness implemented in battle.engine.js:BattlePokemon#isGrounded
+		onBeforeMovePriority: 6,
+		onBeforeMove(pokemon, target, move) {
+			if (move.flags['gravity'] && !move.isZ) {
+				this.add('cant', pokemon, 'move: Gravity', move);
+				return false;
+			}
+		},
+		onModifyMove(move, pokemon, target) {
+			if (move.flags['gravity'] && !move.isZ) {
+				this.add('cant', pokemon, 'move: Gravity', move);
+				return false;
+			}
+		},
+		onFieldResidualOrder: 27,
+		onFieldResidualSubOrder: 2,		
+		onFieldEnd() {
+			this.add('-fieldend', 'move: Gravity');
+			const leftstate = this.p1.active[0];
+			const rightstate = this.p2.active[0];
+			const lunellonleft = (leftstate && leftstate.isActive && leftstate.name == "Lunell");
+			const lunellonright = (rightstate && rightstate.isActive && rightstate.name == "Lunell");
+			if (lunellonleft || lunellonright) {
+				this.add(`c:|${getName('Lunell')}|ope there goes gravity`);
+			}			
```

Comment:
```suggestion
			const activePokemon = this.getAllActive();
			for (let a of activePokemon) {
				if (a.name === "Lunell") {
					this.add(`c:|${getName('Lunell')}|ope there goes gravity`);
					break;
				}
			}
```
This is a cleaner way to implement (unless I'm wrong about `this` being an object of `Battle`).